webui/src/lib/components/repository/changes.jsx

```diff
@@ -117,20 +116,35 @@ export const TreeEntryPaginator = ({ path, setAfterUpdated, nextPage, depth=0, l
     );
 };
 
-function treeItemType(entry, repo, leftDiffRefID, rightDiffRefID) {
-    if (entry.path_type === "object") {
-        return TreeItemType.Object;
-    }
+function useTreeItemType(entry, repo, leftDiffRefID, rightDiffRefID) {
+    const [treeItemType, setTreeItemType] = useState({type: null, loading: true});
 
     // Tree items that represent prefixes are always of entry.type = prefix_changed and the actual diff type is
     // presented at the object level. Therefore, in case of tables that were added or removed we don't know
     // under which of the diff refs the table root is expected to be listed and therefore we try to get the table type
     // from both and take the one that returned results.
-    const tableTypeFromRight = tablesUtil.getTableType(entry.path, repo, rightDiffRefID);
-    const tableTypeFromLeft = tablesUtil.getTableType(entry.path, repo, leftDiffRefID);
-    const tableType = tableTypeFromRight !== null ? tableTypeFromRight : tableTypeFromLeft;
-    if (tableType !== null) {
-        return tablesUtil.tableTypetoTreeItemType(tableType);
-    }
-    return TreeItemType.Prefix;
+    let leftResult = useAPI(() => {
+        if (entry.path_type === "object") {
+            return
+        }
+        return tablesUtil.isDeltaLakeTable(entry.path, repo, rightDiffRefID)
+    });
+    let rightResult = useAPI(() => {
+        if (entry.path_type === "object") {
+            return
+        }
+        return tablesUtil.isDeltaLakeTable(entry.path, repo, leftDiffRefID);
+    });
```

Comment:
```suggestion
    const leftResult = useAPI(() => tablesUtil.isDeltaLakeTable(entry, repo, rightDiffRefID));
    const rightResult = useAPI(() => tablesUtil.isDeltaLakeTable(entry, repo, leftDiffRefID));
```

And move the condition to the inside of `isDeltaLakeTable`.

Reply:
done